src/Engine/MapController.java

```diff
@@ -64,6 +64,10 @@ public void handleInput(InputHandler.InputAction input)
 			default:
 				System.out.println("Invalid InputMode in MapController! " + inputMode);
 		}
+		myGame.currentMenu = null;
+		if (inputMode == InputMode.ACTIONMENU || inputMode == InputMode.PRODUCTION) {
+			myGame.currentMenu = actionMenu;
```

Comment:
Rather than set this here, I'd probably set it in GameInstance wherever the menu is created/deleted.